### PR TITLE
Car-reservation-front-end

### PR DESCRIPTION
**Pull request by yodit93**
### The following were implemented:
- [ ] created properly authenticated login and sign-up pages, while the user,
- registered to the app using username, email, and password,
- logs in to the website, only by typing the username and password,
- [ ] created the navigation panel, in which the user can see links to:
- Models
- Reserve car
- My reservations
- Add car
- Delete car
- [ ] created the main page in which,
- the user can see a list of cars,
- when the user selects a specific car, they can see the car details page,
- [ ] created a car details page in which the user,
- can see the whole details about each car,
- can see a reserve button that takes the user to the reserve form page.
- [ ] created a form for adding the car
- [ ] the "Add Car" link in the navigation panel linked to the form to add a car,
- [ ] the "MyReservation" link
- displays all the reservations of the current user,
- with information about car name, date and city
- [ ] the "Delete Car" link in the navigation panel,
- takes the user to a list of cars added by the current user,
- can see a list of all cars with title and "Delete" button,
- with a corresponding delete button,
- clicking the delete button removed the car from the list.
- [ ] created two different routes for the Reserve form,
1. the reserve car link from the navigation,
- takes the user to the reserve form,
- the user can fill in the start date, the end date, the city and select the car,
- can submit the form and reserve a car.
2, the reserve link from the car details page with a dynamic route,
- takes the user also to the reserve form,
- the user can fill in the start date, the end date, and the city,
- the car is auto-selected using the dynamic route,
- the user can submit the form and reserve a car.
- [ ] used redux for state management,
- [ ] created responsive layouts.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import DeleteCar from './pages/DeleteCar';
+import Models from './pages/Models';
+import MyReservations from './pages/MyReservations';
+import ReserveCar from './pages/ReserveCar';
+import SignUp from './Components/Authentication/SignUp';
+import LogIn from './Components/Authentication/LogIn';
+import PrivateRoutes from './utils/PrivateRoutes';
+import DetailsPage from './pages/DetailsPage';
+
+function App() {
```
Comment (Meltrust) [inline src/App.js:13]: - [ ] When I add a new car, I am presented with the "username" field.  Maybe it should be car name?

image

image

I end up with a car that is named as my username.


Please, rename that field in your form so it is more consistent (like car name). 👍

Comment (Meltrust) [inline src/App.js:13]: - [ ]  Please, kindly refactor this main `App` component to ES6 arrow function, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

That way, your code will meet the latest standards. 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @yodit93,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:


-  Nice code organization ✔️ 
-  App is working well✔️ 
-  Good readme ✔️ 
-  Beautiful frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/App.js (hunk 1 above)
Comment (Progress2002) [reply src/App.js]: Hello @Meltrust, thanks for the observation, the file has been updated.

Comment (Progress2002) [reply src/App.js]: Hello @Meltrust, thanks for the observation, the file has been updated.

**Review by Meltrust — APPROVED** [review]
#### Hi @yodit93,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Car name field has been implemented✔️ 
- `App` component is now ES6✔️ 
- Great job✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**